### PR TITLE
Add networkFirewallPolicyEnforcementOrder option to google_compute_ne…

### PR DESCRIPTION
**Pull request by jenshonkan84**
fixes https://github.com/hashicorp/terraform-provider-google/issues/13952

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**
  release-note:enhancement
```release-note:enhancement
compute: added `network_firewall_policy_enforcement_order` field to `google_compute_network` resource

```

### CODE REVIEW

**Review by hao-nan-li — CHANGES_REQUESTED** [review]
Could you sign the CLA as mentioned in the comments? Thanks!

**Comment by jenshonkan84** [comment]
> Could you sign the CLA as mentioned in the comments? Thanks!

I have done that. As far as I know. The check mark is green. :)

**Review by hao-nan-li — COMMENTED** [review]
I'm a bit surprised that this addition could cause so many different tests to fail. Could you try to run some of the tests locally?

File: mmv1/products/compute/Network.yaml [hunk 1]
```diff
@@ -215,3 +215,12 @@ properties:
       If the field is not speficied, then a /48 range will be randomly allocated from fd20::/20 and returned via this field.
     immutable: true
     default_from_api: true
+  - !ruby/object:Api::Type::Enum
+    name: 'networkFirewallPolicyEnforcementOrder'
```
Comment (hao-nan-li) [inline mmv1/products/compute/Network.yaml:219]: Is this field used in any of the existing test?

Comment (jenshonkan84) [reply mmv1/products/compute/Network.yaml:219]: No I did not add an example for it. It looked like not all fields where covered. So I skiped it. Should I add it?

**Comment by jenshonkan84** [comment]
> I'm a bit surprised that this addition could cause so many different tests to fail. Could you try to run some of the tests locally?

Me also. I have not touched it. I also have some troubles running the test cases. As my org has restricted us to only use the EU region. But most test wants to do stuff in the US.

File: mmv1/products/compute/Network.yaml (hunk 1 above)
Comment (hao-nan-li) [reply mmv1/products/compute/Network.yaml:219]: I'd recommend to add at least 1 test to cover newly added fields.

**Comment by jenshonkan84** [comment]
I have a hard time running the tests as we have org policy to restrict the use of US regions. But I can see that my new test was successful. And this test looks to be failing on more PR:s: TestAccDataSourceDnsManagedZone_basic.

**Comment by jenshonkan84** [comment]
Ok so I tried to run the two failing tests locally.

TestAccComputeServiceAttachment_serviceAttachmentExplicitProjectsExample:
Fails with:
Error: Error creating ServiceAttachment: googleapi: Error 400: Invalid value for field 'resource.domainNames': ''. Please verify ownership of the PTR domain. See more information at https://cloud.google.com/compute/docs/instances/create-ptr-record#domain_ownership., invalid

Looks like the domain in the test is : gcp.tfacc.hashicorptest.com

So I dont know what to do about that.

TestAccDataSourceDnsManagedZone_basic:

Looks to be failing on all PR:s. Also found: https://github.com/GoogleCloudPlatform/magic-modules/pull/7515

That looks like it is known and some one is trying to fix it.

Sorry if I am misunderstanding something I am new at contributing here. :)

**Comment by hao-nan-li** [comment]
> Ok so I tried to run the two failing tests locally.
> 
> TestAccComputeServiceAttachment_serviceAttachmentExplicitProjectsExample: Fails with: Error: Error creating ServiceAttachment: googleapi: Error 400: Invalid value for field 'resource.domainNames': ''. Please verify ownership of the PTR domain. See more information at https://cloud.google.com/compute/docs/instances/create-ptr-record#domain_ownership., invalid
> 
> Looks like the domain in the test is : gcp.tfacc.hashicorptest.com
> 
> So I dont know what to do about that.
> 
> TestAccDataSourceDnsManagedZone_basic:
> 
> Looks to be failing on all PR:s. Also found: #7515
> 
> That looks like it is known and some one is trying to fix it.
> 
> Sorry if I am misunderstanding something I am new at contributing here. :)

Don't worry about `TestAccDataSourceDnsManagedZone_basic`.

As far as `TestAccComputeServiceAttachment_serviceAttachmentExplicitProjectsExample`, I don't remember seeing it fail in other test runs. Let's run it again here and see what's the outcome.

**Pull request merged**